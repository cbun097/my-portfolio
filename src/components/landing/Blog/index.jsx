import React, { useState, useEffect, useContext } from 'react'
import { ThemeContext } from 'providers/ThemeProvider'
import { Container, Card, Tag, Item } from 'components/common'
import Moment from 'moment'
import { Wrapper, Grid, Content } from './styles'

export const Blog = () => {
  const { theme } = useContext(ThemeContext)
  const [articles, setArticles] = useState([])

  useEffect(() => {
    fetch(`https://dev.to/api/articles?username=cbun097`)
      .then(response => response.json())
      .then(resultData => {
        setArticles(resultData)
      })
  }, [])

  return (
    <Wrapper as={Container} id="blog">
      <h2>Latest post</h2>
      <Grid>
        {articles.map(data => (
          <Item key={data.id} as="a" href={data.url} target="_blank" rel="noopener noreferrer">
            <Card theme={theme}>
              <Content>
                <h3>{data.title}</h3>
                <div>
                  <img src={data.cover_image} alt="article dev" />
                </div>

                <p>{data.description}</p>
              </Content>
              <p>{Moment(data.published_at).format('MMMM Do YYYY')}</p>
              {data.tag_list.map((tag, i) => (
                <Tag key={i}>#{tag}</Tag>
              ))}
            </Card>
          </Item>
        ))}
      </Grid>
    </Wrapper>
  )
}
