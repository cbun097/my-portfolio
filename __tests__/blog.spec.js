import React from 'react'
import renderer from 'react-test-renderer'
import fetchMock from 'fetch-mock'
import { Blog } from '../src/components/landing/Blog/index'

describe('Blog Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Blog />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('fetch data from dev.to', () => {
  it('return successful results', async () => {
    fetchMock.get('https://dev.to/api/articles?username=cbun097', {
      title: 'DevOps Handy Tool: Com2kube',
    })
    const response = await fetch('https://dev.to/api/articles?username=cbun097')
    const result = await response.json()
    expect(result.title).toEqual('DevOps Handy Tool: Com2kube')
  })
})
