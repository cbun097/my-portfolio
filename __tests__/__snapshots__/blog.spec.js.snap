// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Component renders correctly 1`] = `
<div
  className="sc-fzpans cLzOFF sc-fzqBZW iphkXo"
  id="blog"
>
  <h2>
    Latest post
  </h2>
  <div
    className="sc-fzqNJr hFjPfQ"
  />
</div>
`;
